Source/Timeline/AllDayView.swift

```diff
@@ -0,0 +1,163 @@
+
+import UIKit
+
+public protocol AllDayViewDataSource {
```

Comment:
I don't think, there is a need for the `AllDayViewDataSource`, as the `descriptors` can be set in a similar way to `TimelineView`, i.e. as the property.

Also, there is no need for the `numberOfAllDayEvents` call as the number can be calculated by counting the array inside the `AllDayView`.

Reply:
A property would make more sense, I would think because the data doesn't get changed often, perhaps?

> Also, there is no need for the numberOfAllDayEvents call as the number can be calculated by counting the array inside the AllDayView.

Only reason for `numberOfAllDayEvents` to invoke its `dataSource` is because then in `AllDayView.reloadData()`, it asks for a single EventDescriptor at a time. But, if a property were to be used then yes, the protocol would be removed.